src/jdk.incubator.vector/share/classes/jdk/incubator/vector/Byte128Vector.java

```diff
@@ -843,33 +854,101 @@ public ByteSpecies vspecies() {
 
         @Override
         @ForceInline
-        public Byte128Vector toVector() {
-            return VectorSupport.shuffleToVector(VCLASS, ETYPE, Byte128Shuffle.class, this, VLENGTH,
-                                                    (s) -> ((Byte128Vector)(((AbstractShuffle<Byte>)(s)).toVectorTemplate())));
+        Byte128Vector toBitsVector() {
+            return VectorSupport.convert(VectorSupport.VECTOR_OP_REINTERPRET,
+                    Byte128Shuffle.class, byte.class, VLENGTH,
+                    Byte128Vector.class, byte.class, VLENGTH,
+                    this, vspecies().asIntegral(),
+                    (v, s) -> toBitsVectorHelper(v));
+        }
+
+        private static Byte128Vector toBitsVectorHelper(Byte128Shuffle s) {
+            return (Byte128Vector) Byte128Vector.VSPECIES.dummyVector()
+                    .vectorFactory(s.indices());
         }
```

Comment:
Move the implementation details to the super class?

Reply:
I have moved `toBitsVectorTemplate` to `AbstractShuffle`, also `toShuffle` has been refactored to be more versatile.